server/isomorphic-routing/index.js

```diff
@@ -20,22 +20,23 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 
 function combineMiddlewares( ...middlewares ) {
 	return function( req, res, next ) {
-		req.context = getEnhancedContext( req );
+		req.context = getEnhancedContext( req, res );
 		applyMiddlewares( req.context, ...middlewares, () => {
 			next();
 		} );
 	};
 }
 
 // TODO: Maybe merge into getDefaultContext().
-function getEnhancedContext( req ) {
+function getEnhancedContext( req, res ) {
 	return Object.assign( {}, req.context, {
 		isLoggedIn: req.cookies.wordpress_logged_in,
 		isServerSide: true,
 		path: req.url,
 		pathname: req.path,
 		params: req.params,
-		query: req.query
+		query: req.query,
+		res
```

Comment:
Tricky. Not sure if that would make sense. `res` -- being one of the 3 args to an Express mw, alongside `req` and `next`, is quite prominent there, so I think I'd rather not bury it too deep inside something else.

`page` has a `redirect` too, and ideally, we'd make our `redirect` isomorphic. In practice, this is a bit more complex, since it's a member of the `page` object, i.e. `page.redirect()`, as opposed to Express' `res.redirect()`. (Fortuntaly, we don't need the client-side redirect for the legacy Atlas routes, since those always come from outside.)

Inclined to leave as-is. We might try and make things isomorphic to be able to validate filters on the server side, as you suggested elsewhere. But that could well be another iteration :slightly_smiling_face: